src/routes/forms.go

```diff
@@ -0,0 +1,250 @@
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		var resData = dtos.GetFormDetailResponseDto{
+			Id:          form.Id,
+			OwnerId:     form.OwnerId,
+			Status:      string(form.Status),
+			CreatedById: form.CreatedById,
+			UpdatedById: form.UpdatedById,
+			CreatedAt:   form.CreatedAt.String(),
+			UpdatedAt:   form.UpdatedAt.String(),
+			Content:     form.Content,
+			Meta: dtos.GetFormMetaDataResponseDto{
+				Id:                               formMetaData.Id,
+				FormId:                           formMetaData.FormId,
+				IsDeleted:                        formMetaData.IsDeleted,
+				AccepctingResponses:              formMetaData.AccepctingResponses,
+				AllowGuestResponses:              formMetaData.AllowGuestResponses,
+				AllowMultipleRepsonses:           formMetaData.AllowMultipleRepsonses,
+				SendConfirmationEmailToRespondee: formMetaData.SendConfirmationEmailToRespondee,
+				SendSubmissionEmailToOwner:       formMetaData.SendSubmissionEmailToOwner,
+				ValidTill:                        formMetaData.ValidTill,
+				UpdatedById:                      formMetaData.UpdatedById,
+				UpdatedAt:                        formMetaData.UpdatedAt,
+			},
+		}
+
+		var res = dtos.ResponseDto{
+			Message: "form fetched successfully",
+			Data:    resData,
+		}
+
+		ctx.JSON(http.StatusOK, res)
+	})
+
+	forms.PATCH("/:id", func(ctx *gin.Context) {
+		var requestBody dtos.CreateUpdateFormRequestDto
+		if err := ctx.ShouldBindJSON(&requestBody); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "invalid request",
+				Error: &dtos.ErrorResponse{
+					Message: "invalid request body",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		var form models.Form
+		if err := db.NewSelect().Model(&form).Where("id = ?", ctx.Param("id")).Scan(ctx); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "something went wrong",
+				Error: &dtos.ErrorResponse{
+					Message: "error fetching form",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		if requestBody.Status == nil || (*requestBody.Status != string(models.DRAFT) && *requestBody.Status != string(models.PUBLISHED)) {
+			errObj := dtos.ResponseDto{
+				Message: "invalid request",
+				Error: &dtos.ErrorResponse{
+					Message: "invalid status",
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		form.Content = requestBody.Content
+		form.Status = models.FORM_STATUS_TYPE(*requestBody.Status)
+		form.OwnerId = requestBody.PerformedById
+		form.UpdatedById = &requestBody.PerformedById
+
+		if _, err := db.NewUpdate().Model(&form).Where("id = ?", ctx.Param("id")).Exec(ctx); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "something went wrong",
+				Error: &dtos.ErrorResponse{
+					Message: "error updating form",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
```

Comment:
Where are we updating form metadata

Reply:
Working with form mera data is a different set of apis to be covered in a separate PR.